src/parquet/column/writer.h

```diff
@@ -174,12 +183,12 @@ inline void TypedColumnWriter<DType>::WriteBatch(int64_t num_values,
   num_buffered_encoded_values_ += values_to_write;
 
   // TODO(PARQUET-591): Instead of rows as a boundary, do a size check
-  if (num_buffered_values_ >= PAGE_VALUE_COUNT) { WriteNewPage(); }
+  if ((num_buffered_values_ >= PAGE_VALUE_COUNT) && !has_dictionary_) { WriteNewPage(); }
```

Comment:
If/when we do this, we can allocate memory for the encoded/compressed data pages from the common `MemPool` which you added already to this class